.travis.yml

```diff
@@ -1,110 +1,156 @@
+- echo $TOXENV
+- python --version
+- uname -a
+- lsb_release -a
 install:
-  - pip install --upgrade tox setuptools virtualenv wheel
-  - virtualenv --version
-  - easy_install --version
-  - pip --version
-  - tox --version
+- pip install --upgrade tox setuptools virtualenv wheel
+- virtualenv --version
+- easy_install --version
+- pip --version
+- tox --version
 script:
-  - tox -v
+- tox -v
 after_failure:
-  - more .tox/log/* | cat
-  - more .tox/*/log/* | cat
+- more .tox/log/* | cat
+- more .tox/*/log/* | cat
 before_cache:
-  - rm -rf $HOME/.cache/pip/log
+- rm -rf $HOME/.cache/pip/log
 cache:
   directories:
-    - $HOME/.cache/pip
+  - "$HOME/.cache/pip"
 notifications:
   email:
     on_success: never
     on_failure: always
   slack:
     rooms:
-      - secure: nsbTI19SAiog6V6OnN1diYU2V3RIwtmY+IqV4uAvki87dQZteWU22/kQ0wI0OZAOzvEG3lLrc7ZGnbKh+IiH3QJkAwwHgtgC2zdzy6EGkjyEQpeAFUujaEHcRKEIpeAdtNpSUTrPsywdbnT6sO4XN/S7ZhSxS3TRmL2Wi399i36/D0T7gt/GAcPdxrFKwZ20PN20By+5j54FO6/X7y0IVcoUgKdYIvP5NGILRyo+tlWP6dJKFeuxT7lU3FV6W7J7DQs/2yjwH1naBmulxWp22+wNopJ1loW9PHtsJzlN0kUp1Up/+pwHvr7bVbZBzXSZZ5V7j+Xi7uh0de8GYckg0ATJTvrB5c1KZaG9p2unhyMlVAyDfNez3LtgUERnd7S1YXxz8JwbyWlEodzKO/dcnA1PkzcPECEbeXQJtxdhXuz+8pVqIAkDejCv4bhZVUtC4beeQFjxTrqKewxMNpTKKCacIWEBLrCUzr4y15qy7ziRU8CPvFkaUYBkPSWIOp2Xv7FOBwlEhR8d5P6P2eJ7P1vbYVB1vvrYgo0Z4pQkaM02E2/YgSoEfpWsgrSNS+9W0aQtAxcIg6ilY4ZcY83xAmwa8W7uILygqqK0Q7m+UiirIePMWXJ3r9MNkpHHlEr+WebFQYYNhOqVDJ5oX1+KKZY91bJl0ozqUvy4KSlBPO8=
+    - secure: nsbTI19SAiog6V6OnN1diYU2V3RIwtmY+IqV4uAvki87dQZteWU22/kQ0wI0OZAOzvEG3lLrc7ZGnbKh+IiH3QJkAwwHgtgC2zdzy6EGkjyEQpeAFUujaEHcRKEIpeAdtNpSUTrPsywdbnT6sO4XN/S7ZhSxS3TRmL2Wi399i36/D0T7gt/GAcPdxrFKwZ20PN20By+5j54FO6/X7y0IVcoUgKdYIvP5NGILRyo+tlWP6dJKFeuxT7lU3FV6W7J7DQs/2yjwH1naBmulxWp22+wNopJ1loW9PHtsJzlN0kUp1Up/+pwHvr7bVbZBzXSZZ5V7j+Xi7uh0de8GYckg0ATJTvrB5c1KZaG9p2unhyMlVAyDfNez3LtgUERnd7S1YXxz8JwbyWlEodzKO/dcnA1PkzcPECEbeXQJtxdhXuz+8pVqIAkDejCv4bhZVUtC4beeQFjxTrqKewxMNpTKKCacIWEBLrCUzr4y15qy7ziRU8CPvFkaUYBkPSWIOp2Xv7FOBwlEhR8d5P6P2eJ7P1vbYVB1vvrYgo0Z4pQkaM02E2/YgSoEfpWsgrSNS+9W0aQtAxcIg6ilY4ZcY83xAmwa8W7uILygqqK0Q7m+UiirIePMWXJ3r9MNkpHHlEr+WebFQYYNhOqVDJ5oX1+KKZY91bJl0ozqUvy4KSlBPO8=
     on_success: never
-    on_failure: always
+    on_failure: always  
+jobs:
+  include:
+  - stage: check
+    python: 3.6
+    env:
+    - TOXENV=check
+    install:
+    - pip install --upgrade tox setuptools virtualenv wheel
+    script:
+    - tox -v
+    after_failure:
+    - more .tox/log/* | cat
+    - more .tox/*/log/* | cat
+  - stage: docs
+    python: 3.6
+    env:
+    - TOXENV=docs
+    install:
+    - pip install --upgrade tox setuptools virtualenv wheel
+    script:
+    - tox -v
+    after_failure:
+    - more .tox/log/* | cat
+    - more .tox/*/log/* | cat
+  - stage: deploy
+    if: branch in (master, develop)
+    python: 2.7
+    install: pip install m2r
+    script: echo "Deploying to PyPI ..."
+    before_deploy:
+    - if [ "$TRAVIS_BRANCH" = "develop" ]; then
```

Comment:
I _think_ this needs to be written as a multiline string?

```
- |	
  if [ "$TRAVIS_BRANCH" = "develop" ]	
  then	
    sed -i -E "s/^(version = )(.*)/\1\2.$TRAVIS_BUILD_NUMBER/" $TRAVIS_BUILD_DIR/setup.cfg
  fi
```

Reply:
Also, if we aren't going to use travis_set_build.py anymore, then please delete the file...

Reply:
I didn't purposely use a YAML formatter but when I used Travis's own tool to add something (the S3 stuff where I discovered undocumented attributes), it did some interesting things like remove the pipes from multiline things and take the spaces out from lists. I figured Travis was giving it as some sort of recommendation so I left it. Happy to put it back though.